crates/sui-types/src/transaction_displays.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::transaction::CallArg::Pure;
+use crate::transaction::{
+    write_sep, Argument, CallArg, Command, ObjectArg, ProgrammableMoveCall, ProgrammableTransaction,
+};
+use std::fmt::{Display, Formatter};
+use tabled::{
+    builder::Builder as TableBuilder,
+    settings::{style::HorizontalLine, Panel as TablePanel, Style as TableStyle},
+};
+
+pub struct Pretty<'a, T>(pub &'a T);
+
+impl<'a> Display for Pretty<'a, ProgrammableTransaction> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let mut builder = TableBuilder::default();
+
+        let Pretty(ptb) = self;
+        let ProgrammableTransaction { inputs, commands } = ptb;
+
+        for (i, input) in inputs.iter().enumerate() {
+            match input {
+                Pure(v) => {
+                    if v.len() <= 16 {
+                        builder.push_record(vec![format!("{i:<3} Pure Arg          {:?}", v)]);
+                    } else {
+                        builder.push_record(vec![format!(
+                            "{i:<3} Pure Arg          [{}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, ...]",
+                            v[0],
+                            v[1],
+                            v[2],
+                            v[3],
+                            v[4],
+                            v[5],
+                            v[6],
+                            v[7],
+                            v[8],
+                            v[9],
+                            v[10],
+                            v[11],
+                            v[12],
+                            v[13],
+                            v[14],
+                        )]);
+                    }
+                }
+                CallArg::Object(ObjectArg::ImmOrOwnedObject(o)) => {
+                    builder.push_record(vec![format!("{i:<3} Imm/Owned Object  ID: {}", o.0)]);
+                }
+                CallArg::Object(ObjectArg::SharedObject { id, .. }) => {
+                    builder.push_record(vec![format!("{i:<3} Shared Object     ID: {}", id)]);
+                }
+                CallArg::Object(ObjectArg::Receiving(o)) => {
+                    builder.push_record(vec![format!("{i:<3} Receiving Object  ID: {}", o.0)]);
+                }
+            };
+        }
+
+        let mut table = builder.build();
+        table.with(TablePanel::header("Input Objects"));
+        table.with(TableStyle::rounded().horizontals([HorizontalLine::new(
+            1,
+            TableStyle::modern().get_horizontal(),
+        )]));
+        write!(f, "\n{}\n", table)?;
+
+        let mut builder = TableBuilder::default();
```

Comment:
Nit: might be able to simply call `builder.clear()` instead of creating a new one. However, I haven't tried this one so not sure how it works with styling.